controllers/modelmesh/runtime.go

```diff
@@ -333,3 +333,32 @@ func (m *Deployment) addPassThroughPodFieldsToDeployment(deployment *appsv1.Depl
 
 	return nil
 }
+
+func (m *Deployment) configureRuntimePodSpecAnnotations(deployment *appsv1.Deployment) error {
+
+	if deployment.Spec.Template.Annotations == nil {
+		deployment.Spec.Template.Annotations = make(map[string]string)
```

Comment:
Small thing but for these you could just have this be
```go
    deployment.Spec.Template.Annotations = m.AnnotationsMap
```
and then have the loop below in an `else` block. Same for the labels.